### PR TITLE
Integrate Prettier into eslint and stylelint workflow

### PR DESCRIPTION
**Pull request by spmcbride1201**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          key: dependency-cache-{{ checksum "package.json" }}
+      - run:
+          name: install-npm
+          command: npm install
+      - save_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - .node_modules
+
+  test:
```
Comment (dgoldstein1) [inline .circleci/config.yml]: `npm test` does not pass, but same error occurs for `master` so probably not something related to these changes.

File: .eslintrc [hunk 2]
```diff
@@ -1,41 +1,11 @@
 {
   "parser": "babel-eslint",
   "rules": {
```
Comment (dgoldstein1) [inline .eslintrc]: linting looks good here :) and correspond to the linting changes below

File: .vscode/cSpell.json [hunk 3]
```diff
@@ -29,7 +29,25 @@
...
+        "precached",
+        "linebreak",
+        "Finalizer",
+        "multilines",
+        "Memoized",
+        "Resizable",
+        "precommit"
     ],
     // flagWords - list of words to be always considered incorrect
     // This is useful for offensive words and common spelling errors.
```
Comment (dgoldstein1) [inline .vscode/cSpell.json]: This is really cool --> Is this an extension for virtual studio?

File: package.json [hunk 4]
```diff
@@ -50,23 +53,27 @@
...
-    "eject": "react-scripts eject"
+    "watch": "npm-watch"
+  },
+  "lint-staged": {
+    "*.js": ["prettier --write", "git add"],
+    "*.less": ["stylelint --fix", "git add"],
+    "*.css": ["stylelint --fix", "git add"]
   },
   "eslintConfig": {
     "extends": "react-app"
```
Comment (dgoldstein1) [inline package.json]: 👍

File: src/components/Explorer.js [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
+import { connect } from "react-redux";
+import _ from "lodash";
+import { Responsive, WidthProvider } from "react-grid-layout";
+import GMLineChart from "./GMLineChart";
+import { getTimeSeriesOfValue } from "../utils";
 const ResponsiveReactGridLayout = WidthProvider(Responsive);
 
 class Explorer extends Component {
   static propTypes = {
+    keys: PropTypes.array,
```
Comment (dgoldstein1) [inline src/components/Explorer.js]: I don't understand what this change is doing

File: src/components/golang/SummaryBar.js [hunk 6]
```diff
@@ -1,55 +0,0 @@
...
-      </div >
-    </div >
-  );
-}
-
-function mapStateToProps({ metrics: { memory } }) {
-  return { memory };
-};
-
-export default connect(mapStateToProps)(SummaryBar);
```
Comment (dgoldstein1) [inline src/components/golang/SummaryBar.js]: Why was this deleted? Was it replaced with something else?

File: .circleci/config.yml (hunk 1 above)
Comment (spmcbride1201) [inline .circleci/config.yml]: Very interesting. Seems to pass for me. 
image

File: .vscode/cSpell.json (hunk 3 above)
Comment (spmcbride1201) [inline .vscode/cSpell.json]: Yep, it's sort of goofy, but I kept making spelling mistakes in my comments, so I've given it a shot. It's sort of add that it inserts comments into JSON, but it works.

File: src/components/golang/SummaryBar.js (hunk 6 above)
Comment (spmcbride1201) [inline src/components/golang/SummaryBar.js]: It's effectively dead code dating from before my redesign to create dashboards from JSON.

File: src/components/Explorer.js (hunk 5 above)
Comment (spmcbride1201) [inline src/components/Explorer.js]: My eslint rules require PropTypes to be set for all props to provide basic type checking. PropTypes validates that keys is of type array and errors if something else is passed in.

File: src/components/golang/SummaryBar.js (hunk 6 above)
Comment (dgoldstein1) [inline src/components/golang/SummaryBar.js]: I was confused because I didn't see a commit 'deleting dead code'

File: .circleci/config.yml (hunk 1 above)
Comment (dgoldstein1) [inline .circleci/config.yml]: On review the issue was `watchman` was not installed. Tests are passing after installation.

File: src/components/golang/SummaryBar.js (hunk 6 above)
Comment (spmcbride1201) [inline src/components/golang/SummaryBar.js]: You're right. I've gotten sloppy. I'll be more mindful of this going forward.
